cloudferrylib/os/actions/check_networks.py

```diff
@@ -59,6 +75,40 @@ def __init__(self, info):
     def get_networks(self):
         return self.by_hash.values()
 
+    def get_segmentation_ids(self):
+        """Get busy segmentation IDs.
+
+        We need to handle duplicates in segmentation ids.
+        Neutron has different validation rules for different network types.
+
+        For 'gre' and 'vxlan' network types there is no strong requirement
+        for 'physical_network' attribute, if we want to have
+        'segmentation_id', because traffic is encapsulated in L3 packets.
+
+        For 'vlan' and 'flat' network types there is a strong requirement for
+        'physical_network' attribute, if we want to have 'segmentation_id'.
+
+        :result: Dictionary with busy segmentation IDs.
+                 Hash is used with structure {"gre": [1, 2, ...],
+                                              "vlan": [1, 2, ...]}
+        """
+
+        used_seg_ids = {}
+        networks = self.by_id.values()
+
+        for net in networks:
+            network_has_segmentation_id = (
+                net.info["provider:physical_network"] or
+                (net.network_type in ['gre', 'vxlan']))
```

Comment:
No. For `VLAN` there is a strong requirement for `provider:physical_network` attribute, if we want to have `provider:segmentation_id`. So in this case we are fine with the first condition. I've already specified it in the docstring above.